### PR TITLE
MDL-78528 - [docs] Append a suffix to the completion rules

### PR DESCRIPTION
**Pull request by sarjona**
This patch adds information about the changes done in [MDL-78528](https://tracker.moodle.org/browse/MDL-78528).

<a href="https://gitpod.io/#https://github.com/moodle/devdocs/pull/699"></a>

### CODE REVIEW

**Review by andrewnicols — APPROVED** [review]

File: docs/devupdate.md [hunk 1]
```diff
@@ -334,3 +334,48 @@ if (!empty($showonly)) {
...
+
+As part of [MDL-78516](https://tracker.moodle.org/browse/MDL-78516), the [Default completion form](https://docs.moodle.org/en/Activity_completion_settings#Changing_activity_completion_settings_in_bulk) has undergone a significant rebuild to enhance code reusability and maintainability. To prevent duplicate IDs, a suffix has been introduced to the form elements related to completion rules.
+
+For third-party plugins, an adjustment is needed to incorporate this new suffix, following the approach already taken by [core modules](https://github.com/sarjona/moodle/commit/8f57f0fdaca027c7099bc6966467077aecbc0862).
+
+The primary modification entails editing `mod/yourplugin/mod_form.php` and applying the suffix to the completion rule elements within all relevant methods. As an example, here are the changes made to the `mod/choice` module:
+
+```php
+    public function add_completion_rules() {
+        $mform =& $this->_form;
```
Comment (andrewnicols) [inline docs/devupdate.md:350]: Can we remove the & from this. It's totally unnecessary :)

File: docs/devupdate.md [hunk 2]
```diff
@@ -334,3 +334,48 @@ if (!empty($showonly)) {
...
+
+        $suffix = $this->get_suffix();
+        $completionsubmitel = 'completionsubmit' . $suffix;
```
Comment (andrewnicols) [inline docs/devupdate.md:353]: Can I suggest that we replace this duplicated string concate with a function:

```
$this->get_suffixed_name('completionsubmit');

protected function get_suffixed_name(string $fieldname): string {
    return $fieldname . $this->get_suffix();
}
```

The suffix concat is duplicated on half a dozen lines and we should aim to make our examples more readable.

File: docs/devupdate.md (hunk 1 above)
Comment (sarjona) [reply docs/devupdate.md]: You're right!! Well spotted! :-)

File: docs/devupdate.md (hunk 2 above)
Comment (sarjona) [reply docs/devupdate.md]: That's a great idea. I'll see if I can add this method to the trait in order to call it! ;-) If that's the case, I'll update again the docs in the future (but, for now, I've added it too).

**Review by andrewnicols — APPROVED** [review]

**Pull request merged**